docs/recipes/probes.md

```diff
@@ -0,0 +1,172 @@
+
+* `Started` - A replica is in this state when it has just been started, and hasn't been probed yet.  
+* `Healthy` - A replica is in this state when it passes the `liveness` probe, but not the `readiness` probe. 
+* `Ready` - A replica is in this state when it passes both the `liveness` and `readiness` probes.  
+
+(*Internally, there are more states that a replica can be in, but those are not relevant to this discussion*)
+
+## Types of Probes  
+
+There are two types of probes, both have a similar schema but serve a different purpose.  
+
+### Liveness  
+
+The `liveness` probe is used to let Tye know when it can restart a replica.  
+
+When a replica is restarted due to a failed `liveness` probe, a new replica is created in its stead.
+
+### Readiness  
+
+The `readiness` probe is used to let Tye know when it's okay to route traffic to a replica.  
+
+The `readiness` probe cannot kill/restart a replica, it can only do two things
+
+* Promote `Healthy` replica to `Ready`, when the probe succeeds.
+* Demote a `Ready` replica to `Healthy` when the probe fails.  
+
+Tye only routes traffic (either via a service binding, or an ingress) to `Ready` replicas.  
+
+## Life cycle of a Replica in the Absence of Probes
+
+By now, the life cycle of a replica when both `liveness` and `readiness` probes are configured should be clear, but how does the life cycle of a replica look like when both probes or one of the probes is absent?  
+
+* When both `liveness` and `readiness` probes are absent, a replica gets promoted from `Started` to `Ready` automatically, upon creation.
+* When only the `readiness` probe is absent, a replica gets promoted from `Started` to `Healthy` only after it passes the `liveness` probe, but upon being promoted to `Healthy`, it's automatically promoted again, to `Ready`.  
+* When only the `liveness` probe is absent, a replica gets promoted from `Started` to `Healthy` automatically, upon creation, but gets promoted from `Healthy` to `Ready` only upon passing the `readiness` probe.  
+
+## Running Locally with Liveness and Readiness Probes
+
+*The section will refer to this [sample application](/samples/liveness-and-readiness/) which demonstrates basic usage.*  
+
+To run the sample locally, clone this repository or download the source and navigate to the `/samples/liveness-and-readiness/` directory in your terminal.  
+
+```
+tye run
+```
+
+The sample has a single service with a `liveness` and a `readiness` probe, as shown in this snippet
+
+```
+services:
+  - name: simple-webapi
+    project: webapi/webapi.csproj
+    replicas: 2
+    liveness:
+      http:
+        path: /healthy
+    readiness:
+      http:
+        path: /ready
+```
+
+The service is configured to respond successfully to both the `liveness` and `readiness` probes, so after you run Tye, you should see these log lines  
+
+```
+[18:14:05 INF] Replica simple-webapi_a2d67bd9-4 is moving to an healthy state
```

Comment:
Are these logs now debug level? If we change the tye run command to not log debug by default, make sure to specify the log level to debug.

Reply:
These logs were `debug` before this PR, but were changed to `info`

Reply:
Ah gotcha.